packages/opentelemetry/src/utils/dynamicSamplingContext.ts

```diff
@@ -0,0 +1,61 @@
+import { TraceFlags } from '@opentelemetry/api';
+import {
+  SEMANTIC_ATTRIBUTE_SENTRY_SAMPLE_RATE,
+  SEMANTIC_ATTRIBUTE_SENTRY_SOURCE,
+  getClient,
+  getDynamicSamplingContextFromClient,
+} from '@sentry/core';
+import type { DynamicSamplingContext } from '@sentry/types';
+import { baggageHeaderToDynamicSamplingContext } from '@sentry/utils';
+import { SENTRY_TRACE_STATE_DSC } from '../constants';
+import type { AbstractSpan } from '../types';
+import { spanHasAttributes, spanHasName } from './spanTypes';
+
+/**
+ * Creates a dynamic sampling context from a span (and client and scope)
+ *
+ * @param span the span from which a few values like the root span name and sample rate are extracted.
+ *
+ * @returns a dynamic sampling context
+ */
+export function getDynamicSamplingContextFromSpan(span: AbstractSpan): Readonly<Partial<DynamicSamplingContext>> {
+  const client = getClient();
+  if (!client) {
+    return {};
+  }
+
+  const dsc = getDynamicSamplingContextFromClient(span.spanContext().traceId, client);
+
+  const traceState = span.spanContext().traceState;
+  const traceStateDsc = traceState?.get(SENTRY_TRACE_STATE_DSC);
+
+  if (traceStateDsc) {
+    const dsc = baggageHeaderToDynamicSamplingContext(traceStateDsc);
+    if (dsc) {
+      return dsc;
+    }
```

Comment:
`dsc` is defined twice in this scope. Can we somehow pull them apart so that it is more clear what the order of priority is in this function? I personally would first do the base-case and check whether there is a tracestate/dsc on the span and return that one. If it's not there we should do the rest of the logic.

Reply:
good point, will rewrite this a bit!